lib/server.js

```diff
@@ -68,53 +47,87 @@ function createServer (options, httpHandler) {
       } else {
         /* Deal with listen ([port[, host[, backlog]]]) */
         options.port = argsLength >= 1 && firstArg ? firstArg : 0
-        // This will listen to what localhost is.
-        // It can be 127.0.0.1 or ::1, depending on the operating system.
-        // Fixes https://github.com/fastify/fastify/issues/1022.
-        options.host = argsLength >= 2 && args[1] ? args[1] : 'localhost'
+        options.host = host
         options.backlog = argsLength >= 3 ? args[2] : undefined
       }
 
       return options
     }
 
+    const bindedListen = _listen.bind(this)
     const listenOptions = normalizeListenArgs(Array.from(arguments))
+
+    let secondHost
+    let clonedOptions
+    // TODO: Remove once finished
+    process._rawDebug('Binding second host:', listenOptions.host === '127.0.0.1' || listenOptions.host === '::1')
+    if (listenOptions.host === '127.0.0.1' || listenOptions.host === '::1') {
```

Comment:
I think the check should be on `'localhost'`. If somebody specifies `127.0.0.1` or `::1`, we should honor that.

Reply:
Agreed.

Reply:
Sounds good to me!